core/store/src/archive/cold_storage.rs

```diff
@@ -475,12 +479,15 @@ fn get_keys_from_store(
                         DBCol::OutcomeIds.key_type(),
                         &[DBKeyType::BlockHash, DBKeyType::ShardId]
                     );
-                    shard_layout
-                        .shard_ids()
-                        .map(|shard_id| {
+                    tracked_shards
```

Comment:
It's not immediately obvious to me that we shouldn't use all shards here and for other key types. It's true that we only track a subset of shards, but does that imply that the other shard ids are not used in the db keys for some data? For example some shards that we track might have received some receipts from other shards - how is that stored in the db? 

I think the easiest thing to do would be to use all shards here. The hard thing to do would be to convince ourselves that this implementation does not lose any data when copying hot->cold and that it is worth doing. Making it future proof is harder yet.